env/env_test.cc

```diff
@@ -459,8 +460,11 @@ TEST_P(EnvPosixTestWithParam, RunMany) {
   env_->Schedule(&CB::Run, &cb3);
   env_->Schedule(&CB::Run, &cb4);
 
-  Env::Default()->SleepForMicroseconds(kDelayMicros);
-  int cur = last_id.load(std::memory_order_acquire);
+  int cur = 0;
+  for (int i = 0; i < kNumDelayAllowed && cur != 4; i++) {
+    Env::Default()->SleepForMicroseconds(kDelayMicros);
+    cur = last_id.load(std::memory_order_acquire);
+  }
```

Comment:
Why not wait for the thread pools to be empty and then check cur?  Isn't this essentially just making sure the threads are scheduled in the right order and not so much caring the intermediate results?